packages/strapi-generate-api/test/endpoint.test.e2e.js

```diff
@@ -679,180 +579,142 @@ describe('Test oneToOne relation (article - reference) with Content Manager', ()
+        url: '/tags',
         method: 'POST',
-        json: true,
         body: {
-          name: 'cat111',
-          tag: tagToCreate
-        }
-      });
-
-      expect(referenceToCreate.tag.id).toBe(tagToCreate.id);
-    }
-  );
-
-  test(
-    'Detach Tag to a Reference',
-    async () => {
-      const tagToCreate = await rq({
-        url: `/tags`,
+          name: 'tag111',
+        },
+      }).then(({ body: tagToCreate }) => {
+        return rq({
+          url: '/references',
+          method: 'POST',
+          body: {
+            name: 'cat111',
+            tag: tagToCreate,
+          },
+        }).then(({ body }) => {
+          expect(body.tag.id).toBe(tagToCreate.id);
+        });
+      });
+    });
+
+    test('Detach Tag to a Reference', async () => {
+      const { body: tagToCreate } = await rq({
+        url: '/tags',
         method: 'POST',
-        json: true,
         body: {
-          name: 'tag111'
-        }
+          name: 'tag111',
+        },
       });
 
-      const referenceToCreate = await rq({
-        url: `/references`,
+      const { body: referenceToCreate } = await rq({
+        url: '/references',
         method: 'POST',
-        json: true,
         body: {
           name: 'cat111',
-          tag: tagToCreate
-        }
+          tag: tagToCreate,
+        },
       });
 
       expect(referenceToCreate.tag.id).toBe(tagToCreate.id);
 
-      const referenceToUpdate = await rq({
+      const { body: referenceToUpdate } = await rq({
         url: `/references/${referenceToCreate.id}`,
         method: 'PUT',
-        json: true,
         body: {
-          tag: null
-        }
+          tag: null,
+        },
       });
 
       expect(referenceToUpdate.tag).toBe(null);
-    }
-  );
-
-  test(
-    'Delete Tag so the relation in the Reference side should be removed',
-    async () => {
-      const tagToCreate = await rq({
-        url: `/tags`,
+    });
+
+    test('Delete Tag so the relation in the Reference side should be removed', async () => {
+      const { body: tagToCreate } = await rq({
+        url: '/tags',
         method: 'POST',
-        json: true,
         body: {
-          name: 'tag111'
-        }
+          name: 'tag111',
+        },
       });
 
-      const referenceToCreate = await rq({
-        url: `/references`,
+      const { body: referenceToCreate } = await rq({
+        url: '/references',
         method: 'POST',
-        json: true,
         body: {
           name: 'cat111',
-          tag: tagToCreate
-        }
+          tag: tagToCreate,
+        },
       });
 
-      const tagToDelete = await rq({
+      await rq({
         url: `/tags/${tagToCreate.id}`,
         method: 'DELETE',
-        json: true
       });
 
-      const referenceToGet = await rq({
-        url: `/references/${referenceToCreate.id}`,
+      const { body: referenceToGet } = await rq({
+        url: `/references/${referenceToCreate.id || referenceToCreate._id}`,
```

Comment:
I suppose you add it to manager mongoose entries.
But `.id` also exist for mongoose nop ?

Reply:
Forgot to remove it